build_darwin_arm64.sh

```diff
@@ -12,7 +12,7 @@ fi
 
 rm -rf arm64 && mkdir arm64
 
-curl -fsSL "https://zlib.net/zlib-$ZLIB_VERSION.tar.gz" -o zlib-$ZLIB_VERSION.tar.gz
```

Comment:
`macOS nodes` are getting blocked by the `zlib.net` URLs. So using a mirror instead.